preprocess_ecearth.py

```diff
@@ -53,373 +23,213 @@
+    return xr.open_dataset(filepath).sel(time=date.strftime("%Y%m%d"))
 
 
-def get_input_data_next_month(variable, year, month):
-    """Get input data for next month(?)"""
-    if month == 12:
-        return get_input_data(variable, year + 1, 1)
+def get_input_data(variable, date):
+    """Workaround for keeping original timestamps crossing midnight."""
+    day1 = _get_input_data(variable, date)
+    day2 = _get_input_data(variable, date + dt.timedelta(days=1))
+    days = xr.concat([day1, day2], dim="time")
+    if len(days.time) < 9:
+        return days.isel(time=slice(0, 5))
     else:
-        return get_input_data(variable, year, month + 1)
+        return days.isel(time=slice(0, 10))
 
 
-def get_output_data(year, month, a):
+def get_output_path(year, month, a):
     """Get data for output file."""
     filename = f"{year}-{month:02d}-{a:02d}fluxes_storages.mat"
     save_path = os.path.join(config["interdata_folder"], filename)
     return save_path
 
 
-# Code (no need to look at this for running)
+def join_levels(pressure_level_data, surface_level_data):
+    """Combine 3d pressure level and 2d surface level data.
 
-# Determine the fluxes and states
-# In this defintion the vertical spline interpolation is performed to determine the moisture fluxes for the two layers at each grid cell
-def getWandFluxes(
-    latnrs,
-    lonnrs,
-    final_time,
-    a,
-    year,
-    month,
-    begin_time,
-    count_time,
-    density_water,
-    latitude,
-    longitude,
-    g,
-    A_gridcell,
-):
+    A dummy value of 1000 hPa is inserted for the surface pressure
+    """
+    bottom = pressure_level_data.isel(lev=0).copy()
+    bottom.values = surface_level_data.values
+    bottom["lev"] = 100000.0  # dummy value
 
-    if a != final_time:  # not the end of the year
-        # specific humidity atmospheric data is 6-hourly (06.00,12.00,18.00, 00.00)
-        q = get_input_data("Q", year, month).variables["Q"][
-            begin_time : (begin_time + count_time + 1), :, latnrs, lonnrs
-        ]  # kg/kg
-        time = get_input_data("Q", year, month).variables["time"][
-            begin_time : (begin_time + count_time + 1)
-        ]
-        q_levels = get_input_data("Q", year, month).variables["lev"][:]
-        # specific humidity surface data is 3-hourly (03.00,06.00,09.00,12.00,15.00,18.00,21.00,00.00)
-        q2m = get_input_data("Q2M", year, month).variables["Q2M"][
-            begin_time * 2 + 1 : (begin_time * 2 + count_time * 2 + 1) + 1 : 2,
-            latnrs,
-            lonnrs,
-        ]  # kg/kg #:267,134:578
-        time_q2m = get_input_data("Q2M", year, month).variables["time"][
-            begin_time * 2 + 1 : (begin_time * 2 + count_time * 2 + 1) + 1 : 2
-        ]
-
-        # surface pressure is 3-hourly data (03.00,06.00,09.00,12.00,15.00,18.00,21.00,00.00)
-        lnsp = get_input_data("LNSP", year, month).variables["LNSP"][
-            begin_time * 2 : (begin_time * 2 + count_time * 2 + 1) : 2,
```

Comment:
Note: this indexing is different from u10, q2m etc:

* q2m: `begin_time * 2 + 1 : (begin_time * 2 + count_time * 2 + 1) + 1 : 2` --> 1:10: 2 = [1, 3, 5, 7, 9]
* lnsp: `begin_time * 2 : (begin_time * 2 + count_time * 2 + 1) : 2` --> 0:9:2 = [0, 2, 4, 6, 8]

where I added how this evaluates for the first day.